### ISSUE TITLE
New GAN (Hifi-gan replacement)

### ISSUE THREAD

**Issue by litsa-the-dancer** [fumiama/Retrieval-based-Voice-Conversion-WebUI#28]
Since hifi-gan is really old and outdated, i suggest you add something newer and more robust such as BIGvae gan. The results seem promising and can improve the overall performance of the model and potentially help with stuff such as mode collapse and noise handling. All in all, it's about time RVC gets a newer and more efficient GAN.  ^w^

**Comment by fumiama** [comment]
Thanks for your advice. I will start doing this after the code being refactored into a programmer-friendly style. Anyone who has interest can also help us by mentioning good algorithms or open PRs.

**Comment by blaisewf** [comment]
I don't think that the change to other gans will generate a substantial change in quality, the current quality of RVC is quite good, where I think there is potential is mainly in the optimization of the new architectures

**Comment by fumiama** [comment]
> I don't think that the change to other gans will generate a substantial change in quality

Yes, the hifigan in RVC is modified so it has already been bound in the algorithm.

> where I think there is potential is mainly in the optimization of the new architectures

Yes, I will do this first. This issue will be considered lastly.

**Comment by ZygoteCode** [comment]
Applio developers are already implementing RVC v3 with BIGVGAN as an experiment in their repository:

https://github.com/IAHispano/Applio/tree/exp-bigvgan

I think we can contribute to that or merge in this repository the changes that they're doing. Idk if the devs here want to contribute with blaise-tk (developer of Applio).

@fumiama @alexlnkp

**Comment by blaisewf** [comment]
> Applio developers are already implementing RVC v3 with BIGVGAN as an experiment in their repository:
> 
> https://github.com/IAHispano/Applio/tree/exp-bigvgan
> 
> I think we can contribute to that or merge in this repository the changes that they're doing. Idk if the devs here want to contribute with blaise-tk (developer of Applio).
> 
> @fumiama @alexlnkp

currently it is a bit broken

**Comment by fumiama** [comment]
I will continue refacturing this repo first. After that, I will consider algorithm improvement.
